src/components/Tabs/Tab.jsx

```diff
@@ -0,0 +1,15 @@
+import cn from 'classnames';
+
+export const Tab = ({ tab, activeTab, onTabChange }) => (
+  <li className={cn({ 'is-active': activeTab.id === tab.id })} data-cy="Tab">
+    <a
+      href={`#${tab.id}`}
+      data-cy="TabLink"
+      onClick={() => {
+        onTabChange(tab);
```

Comment:
Requirements:

7. Pass the onTabSelected callback to the Tabs, it should be called whenever the user selects another tab. (Don't call the callback if the tab was not changed)

So, you shouldn't invoke onTabChange if activeTab is equal to a tab user clicks. 
And move this onClick handler to an own function